src/api/app.py

```diff
@@ -166,120 +173,109 @@ def uploadHandler():
     return session_res
 
+@app.delete('/api/session')
+def log_out(request: Request, session: SessionDeletePydantic):
+    response = session_controller.delete_session(session.dict())
 
-@app.route('/api/session', methods=['DELETE'])
-def log_out():
-    response = session_controller.delete_session(request.json)
-
-    if response.get_json()['success']:
-        session.pop('user', None)
+    if response['success']:
+        request.session.pop('user', None)
 
     return response
 
-
-@app.route('/api/event', methods=['POST'])
-def add_user_event():
-    return event_controller.add_event(json.loads(request.data))
-
-@app.route('/api/user/course', methods=['POST'])
-def add_student_course():
-    info = request.get_json()
-
-    if 'user' not in session:
-        return Response("Not authorized", status=403)
-
-    resp, error = course_select.add_selection(info['name'], info['semester'], session['user']['user_id'], info['cid'])
-    return Response(status=200) if not error else Response(error, status=500)
-
-
-@app.route('/api/user/course', methods=['DELETE'])
-def remove_student_course():
-    info = request.json
-
-    if 'user' not in session:
-        return Response("Not authorized", status=403)
-
-    resp, error = course_select.remove_selection(info['name'], info['semester'], session['user']['user_id'], info['cid'])
-    return Response(status=200) if not error else Response(error, status=500)
-
-@app.route('/api/user/course', methods=['GET'])
-def get_student_courses():
-    if 'user' not in session:
-        return Response("Not authorized", status=403)
-
-    courses, error = course_select.get_selection(session['user']['user_id'])
-    return jsonify(courses) if not error else Response(error, status=500)
-
-if __name__ == '__main__':
-    app.run(debug=os.environ.get('DEBUG', 'True'), host='0.0.0.0', port=5000)
+@app.post('/api/event')
+def add_user_event(request: Request, credentials: SessionPydantic):
+    return Response(status_code=501)
+    #return event_controller.add_event(json.loads(request.data))
+
+@app.post('/api/user/course')
+async def add_student_course(request: Request, credentials: UserCoursePydantic):
+    if 'user' not in request.session:
+        return Response("Not authorized", status_code=403)
+    resp, error = course_select.add_selection(credentials.name, credentials.semester, request.session['user']['user_id'], credentials.cid)
+    return Response(status_code=200) if not error else Response(error, status_code=500)
+
+
+@app.delete('/api/user/course')
+async def remove_student_course(request: Request, courseDelete:CourseDeletePydantic):
+    if 'user' not in request.session:
+        return Response("Not authorized", status_code=403)
+    resp,error = course_select.remove_selection(courseDelete.name, courseDelete.semester, request.session['user']['user_id'], courseDelete.cid)
+    # resp, error = course_select.remove_selection(info['name'], info['semester'], session['user']['user_id'], info['cid'])
```

Comment:
```suggestion
    # resp, error = course_select.remove_selection(info['name'], info['semester'], session['user']['user_id'], info['cid'])
```
Unused code should be removed.